cli/pkg/kctrl/cmd/app/release/app_spec_builder.go

```
@@ -131,3 +141,16 @@ func (b *AppSpecBuilder) Build() (kcv1alpha1.AppSpec, error) {
 
 	return appSpec, nil
 }
+
+func (b *AppSpecBuilder) checkForErrorsAfterReconciliation(app kcv1alpha1.App, fakeClient *fakekc.Clientset) error {
+	existingApp, err := fakeClient.KappctrlV1alpha1().Apps(app.Namespace).Get(context.Background(), app.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// TODO: How can we prevent reconciler from trying to prepare kapp?
+	if existingApp.Status.UsefulErrorMessage != "" && !strings.Contains(existingApp.Status.UsefulErrorMessage, "Preparing kapp") {
```

Looks like kapp-ctrl reconciler tries to "prepare kapp"([here](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/pkg/deploy/factory.go#L59-L80)) even though `ReleaseCmdRunner` ignores kapp. This ensures that the an error thrown due to the absence of a service account does not make it fail.

Choosing to ignore this error in favour of ensuring release is not dependent on a service account on the cluster even though `kapp` is not being used.

---

I will have to take a closer look at how the reconciler works to be able to suggest something.

---

How an instance of `kapp` is created is defined at a Factory level in kapp-ctrl
So we cannot propagate values "up the tree"

I can imagine a boolean which disables deploy which can be passed to the constructor used to create App CR instances. But that does not make a lot of sense from a kapp-ctrl perspective and would be a change solely for `kctrl`.

---

Sticking with this for now!